### PR TITLE
Bump orient version to 2.2

### PR DESCRIPTION
**Pull request by velo**

### CODE REVIEW

**Comment by velo** [comment]
@mpollmeier hey, do you have the means to summon luca here?

**Comment by mpollmeier** [comment]
On travis it fails with an OutOfMemory error, or is there anything else I missed?

```
Running org.apache.tinkerpop.gremlin.orientdb.gremlintest.structure.OrientGraphStructureStandardIT
Exception in thread "Thread-8" java.lang.OutOfMemoryError: Java heap space
```

That's something we should be able to fix, probably with a change in `.travis.yml`?


I just ran `mvn install` locally and only get one error which looks easy to fix: 

```
searchMatching(org.apache.tinkerpop.gremlin.orientdb.MatcherStepTest)  Time elapsed: 0.569 sec  <<< FAILURE!
java.lang.AssertionError:
Expected: (map containing ["a"->"josh"] and map containing ["c"->"marko"])
     but: map containing ["a"->"josh"] map was [<a=peter>, <c=marko>]
```

Just mention luca if you have a question. He's very happy about our little driver project here and  typically responds quickly.

**Comment by velo** [comment]
Yes, it does... but on my machine it fails as well.... odd!

@lvca so, should we just give orient 2.2 more memory or this is something we should investigate a little big further?

**Comment by velo** [comment]
I just realized that we give tests 2Gb to run

https://github.com/velo/orientdb-gremlin/blob/c93dbf70db07f2f636a9a865234235c4c9b10b7d/driver/pom.xml#L162

**Comment by wildroco** [comment]
I don't know it is helpful, but I ran 'mvn install' and got same result with mpollmeier.

-----------------------------------------------------------------------------------------------------------------
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.928 sec <<< FAILURE!
searchMatching(org.apache.tinkerpop.gremlin.orientdb.MatcherStepTest)  Time elapsed: 0.573 sec  <<< FAILURE!
java.lang.AssertionError: 
Expected: (map containing ["a"->"josh"] and map containing ["c"->"marko"])
     but: map containing ["a"->"josh"] map was [<a=peter>, <c=marko>]
	at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:20)
        ...

**Comment by velo** [comment]
Ya, same OOM issue

**Comment by lvca** [comment]
Sorry for the delay, I'm on it.

**Comment by lvca** [comment]
I've added in the execution setting this new one that is needed from OrientDB v2.2: `-XX:MaxDirectMemorySize=512g`, otherwise the JVM wont allows to use big space in direct memory.

**Comment by velo** [comment]
Hrmmm, i don't think I have a machine capable of running it :/

**Comment by lvca** [comment]
512g is only an upper limit, I think 2G for the test is enough.

**Comment by velo** [comment]
@lvca no sucess =/

File: tests-scala/build.sbt [hunk 1]
```diff
@@ -2,7 +2,7 @@ name := "orientdb-tp3-test"
 organization := "com.michaelpollmeier"
 version := "1.0.0-SNAPSHOT"
 scalaVersion := "2.11.7"
-val orientDBVersion = "2.1.15"
+val orientDBVersion = "2.2.0"
```
Comment (jCalamari) [inline tests-scala/build.sbt]: Why is this version different than the one in pom? (2.24)

Comment (velo) [inline tests-scala/build.sbt]: I didn't bother updating it since the gremlin suite tests don't pass

**Comment by wildroco** [comment]
@lvca Could you please help this work? I really really want to see this driver bumped to OrientDB 2.2!

**Comment by lvca** [comment]
Sorry guys, I'm clearly the bottleneck on this. I'm forwarding this to somebody else of the OrientDB team.

**Comment by velo** [comment]
that would be nice @lvca 

I ran out of ideas....  no matter what I do there is not enough memory!

**Comment by lvca** [comment]
@tglman and @maggiolo00 from OrientDB Team are looking into it.

**Comment by maggiolo00** [comment]
Hi @velo 

let me try to reproduce it

**Comment by maggiolo00** [comment]
@velo 

Do you run the tests on Windows machine?

**Comment by velo** [comment]
Yes, but the CI is Linux and also fails

**Comment by maggiolo00** [comment]
Now  seems stuck

let me check

**Comment by maggiolo00** [comment]
@velo 

i did try with 512g for Max memory and worked for me with 1 test fail but without OOM 

XX:MaxDirectMemorySize=512g

can you try?

**Comment by velo** [comment]
Wow, I don't think I have access to this kind of hardware.

Also, this seems to be a serious memory leak on 2.2 series. I can run 2.1
tests w/o 512g of memory

**Comment by maggiolo00** [comment]
@velo 

as @lvca  explain, this is only an upper limit. OrientDB does not require that in order to run
I did run the tests on my mac with such config

**Comment by mpollmeier** [comment]
we can try it for debugging purposes (@velo please go ahead)

but I agree with @velo that it's not a good idea to actually set the limit so high. it does look like a memory leak, and if orient actually runs anywhere near that value it will slow down or even blow up. if it actually needs that much in our tests you should really take a heap dump and fix the problem.

**Comment by maggiolo00** [comment]
@mpollmeier 

i agree 2G should be fine, i think i may have found something,
i will keep you updated

**Comment by velo** [comment]
Tests passed on my computer using this humongous amount of memory...
It did allocated 10Gb (o.0)

Still that failed on CI

**Comment by maggiolo00** [comment]
@velo 

I've made [this](https://github.com/mpollmeier/orientdb-gremlin/pull/90) PR that implements the graph.drop.

I've used it in the Test Suite so the storages used in tests are removed in teardown. 
You can set it back the -XX:MaxDirectMemorySize=2g paramenter

**Comment by velo** [comment]
@mpollmeier 
any idea why the reconnect test is fail on linux but passing on windows?!

**Comment by velo** [comment]
NVM, fails everywhere =D

**Comment by maggiolo00** [comment]
@velo 

you did remove entirely the  -XX:MaxDirectMemorySize JVM settings.

Use it with 2g
 -XX:MaxDirectMemorySize=2g

**Comment by velo** [comment]
There is no need, tests are passing now, thanks for the assistance!

**Comment by mpollmeier** [comment]
OrientReconnect test is failing because the graph instance behaviour changed - it's returning cached results instead of failing when the underlying connection drops. Will provide a fix shortly and merge everything to master.

**Comment by mpollmeier** [comment]
just released 3.2.1.1 which upgrades to OrientDB 2.2 :)

**Pull request merged**